### ISSUE TITLE
Add support for trail/route filters (location, difficulty, name)

### ISSUE THREAD

**Issue by kachnitel** [kachnitel/ridetime-server#38]
- Should pass the filter through to TF 
    - `rid::1;search::Pipe`
    - `rid::1;difficulty::4`
    - `difficulty::4,5` is supported
  - Deprecates `listByLocation`
- Add `activitytype::1` for MTB (in app)

**Comment by kachnitel** [comment]
# Filter structure

- Drop Criteria in Constructor, create a generic filter where `getCriteria()` exports it into `Criteria`, `getTrailforksQuery()` creates a TF Query
- See EventFilter
  - Rename `EventFilter::apply($filter)` to `getCriteria()`
    - `$filters` param should be replaced with filter values set in class (constructor or setter?)
    - Or `apply($filter)` could be the setter for all filters to be kept in a generic form and only applied to Criteria in `getCriteria`
    - `getCriteria` could have `Criteria $criteria` param
  - create a basic filter method like TF Filter::getFilterItem for scalar properties (can return `=` and `IN`) processed result of Criteria::expr() with others that do not directly translate from query will have custom method 
    - check for custom method before returning default `Criteria::expr()-eq/in($key, $value(s));`? 
  - In EventController:
```
- $criteria = Criteria::create()
-           ->orderBy(array('date' => Criteria::ASC))
-           ->setFirstResult(0)
-           ->setMaxResults(20);
- // Setup filter after
```
---
```
+ // Setup filter first
+ $criteria = $filter->getCriteria()
+           ->orderBy(array('date' => Criteria::ASC))
+           ->setFirstResult(0)
+           ->setMaxResults(20);
```

**Comment by kachnitel** [comment]
Filter support implemented in 859a304 (0.5.5)
EventFilter refactor under review
